### DISCUSSION TITLE
Change patch whith midi CC

### DISCUSSION THREAD

**Discussion by tizfra** [GeekFunkLabs/fluidpatcher#41] [Q&A] [answered]
Hi,
I think is possible to change patch whith midi CC on squishbox, but I don't know where I need to put "INC_PATCH".
Can someone give me an example?
Thanks

**Comment by albedozero** [comment]
The way to do this on squishbox is to add a "patch" type router rule to your bank file(s) where you want to be able to change patches with a CC. This is explained near the end of the `router_rules` definition in the Bank Files section of the wiki. I'll have a lesson video on this soon as well. Here's what the rules would look like:
```yaml
-{type:cc, chan: 1, par1: 21, patch: -1}
-{type:cc, chan: 1, par1: 22, patch: 1}
```

**Reply by tizfra** [reply]
I've tried this rules, but when I load bank LCD tells me: `bank load error: mapping values are not allowed here in "<unicode string>"`

Update:
After several tests with the two rules squishbox no longer receives any midi signal.
If I comment that rules and reload bank squishbox starts working again.

**Comment by albedozero** [comment, marked as answer]
The rules need to be inside a `router_rules` keyword. Here's a more detailed example of a [full bank file](https://gist.github.com/albedozero/44263e13e538492db084c6400f3097b8) with router rules that let you change patches using CCs.
